common/src/main/java/org/apache/sedona/common/S2Geography/PolylineGeography.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.sedona.common.S2Geography;
+
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.UnsafeInput;
+import com.esotericsoftware.kryo.io.UnsafeOutput;
+import com.google.common.collect.ImmutableList;
+import com.google.common.geometry.*;
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.logging.Logger;
+
+/** A Geography representing zero or more polylines using S2Polyline. */
+public class PolylineGeography extends S2Geography {
+  private static final Logger logger = Logger.getLogger(PolylineGeography.class.getName());
+
+  private final List<S2Polyline> polylines;
```

Comment:
We define `PolylineGeography` to contain a list of `S2Polyline`s, which is different from the [C++ version](https://github.com/paleolimbot/s2geography/blob/0.2.0/src/s2geography/geography.h#L179-L203). I think this is a reasonable change, as it makes distinguishing between POLYGON and MULTIPOLYGON much easier as long as we ensure that each `S2Polyline` only contains one polygon. WDYT @paleolimbot